coast/index.py

```diff
@@ -0,0 +1,68 @@
+"""Index class."""
+from dask import array
+from dask.distributed import Client
+from .config import config_parser
+from .logging_util import get_slug, debug, info, warn, warning
+from typing import Union
+from pathlib import Path
+from ast import literal_eval
+
+
+def setup_dask_client(
+        workers: int = 2,
+        threads: int = 2,
+        memory_limit_per_worker: str = '2GB'
+):
+    Client(n_workers=workers, threads_per_worker=threads, memory_limit=memory_limit_per_worker)
+
+
+class Indexed:
+    def __init__(self, config: Union[Path, str] = None):
+        """ Configuration init.
+
+            Args:
+                config (Union[Path, str]): path to json config file.
+        """
+        debug(f"Creating a new {get_slug(self)}")
+
+        self.dataset = None
+        self.chunks = None
+        self.var_mapping = None
+        self.dim_mapping = None
+        self.coord_vars = None
+        self.keep_all_vars = False
+
+        if config:
+            self.json_config = config_parser.ConfigParser(config)
+            if self.json_config.config.chunks:
+                self.chunks = literal_eval(self.json_config.config.chunks[0])
+            self.dim_mapping = self.json_config.config.dataset.dimension_map
+            self.var_mapping = self.json_config.config.dataset.variable_map
+            self.coord_vars = self.json_config.config.dataset.coord_var
+            self.keep_all_vars = literal_eval(self.json_config.config.dataset.keep_all_vars)
+
+    def apply_config_mappings(self) -> None:
+        """Applies json configuration and mappings"""
+        # We iterate through each mapping one by one which enables us to catch those variables that do not exist
+
+        if self.dim_mapping is not None:
+            for k in self.dim_mapping:
```

Comment:
Just as a handy python trick, you can do:
```python
for key, value in self.dim_mapping.items():
# Do stuff
```
So then you don't have to re-call  self.dim_mapping[k] to get the value.

Reply:
noted.